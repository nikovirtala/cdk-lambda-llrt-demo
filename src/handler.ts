import { APIGatewayProxyResult, Handler } from "aws-lambda";

export const handler: Handler = async (event, context): Promise<APIGatewayProxyResult> => {
    const data = { event: event, context: context };
    console.log(data);
    return {
        body: JSON.stringify(data),
        statusCode: 200,
    };
};
