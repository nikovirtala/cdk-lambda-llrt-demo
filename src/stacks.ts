import { join } from "node:path";
import { aws_lambda, CfnOutput, Stack, type StackProps } from "aws-cdk-lib";
import { LlrtFunction } from "cdk-lambda-llrt";

import type { Construct } from "constructs";

export class LambdaLlrtStack extends Stack {
    constructor(scope: Construct, id: string, props: StackProps = {}) {
        super(scope, id, props);

        // define resources here...
        const f = new LlrtFunction(this, "LlrtFunction", {
            applicationLogLevelV2: aws_lambda.ApplicationLogLevel.INFO,
            architecture: aws_lambda.Architecture.ARM_64,
            entry: join(import.meta.dirname, "handler.ts"),
            loggingFormat: aws_lambda.LoggingFormat.JSON,
            systemLogLevelV2: aws_lambda.SystemLogLevel.INFO,
        });

        const fUrl = new aws_lambda.FunctionUrl(this, "LlrtFunctionUrl", {
            function: f,
            authType: aws_lambda.FunctionUrlAuthType.NONE,
        });

        new CfnOutput(this, "LlrtFunctionUrl ", { value: fUrl.url });
    }
}
