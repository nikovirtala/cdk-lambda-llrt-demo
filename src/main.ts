import { App } from "aws-cdk-lib";
import { LambdaLlrtStack } from "./stacks.ts";

// for development, use account/region from cdk cli
const devEnv = {
    account: process.env.CDK_DEFAULT_ACCOUNT,
    region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new LambdaLlrtStack(app, "cdk-lambda-llrt-demo-dev", { env: devEnv });
// new LambdaLlrtStack(app, 'cdk-lambda-llrt-demo-prod', { env: prodEnv });

app.synth();
