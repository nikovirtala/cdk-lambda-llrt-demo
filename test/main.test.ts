import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { LambdaLlrtStack } from "../src/stacks";

test("Snapshot", () => {
    const app = new App();
    const stack = new LambdaLlrtStack(app, "test");

    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
});
