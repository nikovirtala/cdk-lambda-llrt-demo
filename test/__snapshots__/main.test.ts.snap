// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot 1`] = `
{
  "Outputs": {
    "LlrtFunctionUrl": {
      "Value": {
        "Fn::GetAtt": [
          "LlrtFunctionUrl7280802F",
          "FunctionUrl",
        ],
      },
    },
  },
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "LlrtFunctionBBA0B80E": {
      "DependsOn": [
        "LlrtFunctionServiceRole63565B1A",
      ],
      "Properties": {
        "Architectures": [
          "arm64",
        ],
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "2c1a8f684964781466edef2555bfe44acfe6c4f4339b405eee22036e8fbba5ce.zip",
        },
        "Handler": "index.handler",
        "LoggingConfig": {
          "ApplicationLogLevel": "INFO",
          "LogFormat": "JSON",
          "SystemLogLevel": "INFO",
        },
        "Role": {
          "Fn::GetAtt": [
            "LlrtFunctionServiceRole63565B1A",
            "Arn",
          ],
        },
        "Runtime": "provided.al2023",
      },
      "Type": "AWS::Lambda::Function",
    },
    "LlrtFunctionServiceRole63565B1A": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "LlrtFunctionUrl7280802F": {
      "Properties": {
        "AuthType": "NONE",
        "TargetFunctionArn": {
          "Fn::GetAtt": [
            "LlrtFunctionBBA0B80E",
            "Arn",
          ],
        },
      },
      "Type": "AWS::Lambda::Url",
    },
    "LlrtFunctioninvokefunctionurl31E6F0AE": {
      "Properties": {
        "Action": "lambda:InvokeFunctionUrl",
        "FunctionName": {
          "Fn::GetAtt": [
            "LlrtFunctionBBA0B80E",
            "Arn",
          ],
        },
        "FunctionUrlAuthType": "NONE",
        "Principal": "*",
      },
      "Type": "AWS::Lambda::Permission",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
